### ISSUE TITLE
It follows that, Ryan Cavanaugh should go with option 2 (prototypes) and Symbols!!

### ISSUE THREAD

**Issue by arturohernandez10** [staeke/ts-extension-methods#1]
Would you take a PR with some change to the readme, suggesting to go with 2. And adding support for it in Typescript such that.

    myDate.addDays(3)
    
Is replaced by:

    myDate[addDays](3)

Of course there is some type and code generation issues to resolve. But It seems like the sensitive thing to do.

**Comment by staeke** [comment]
I agree. There is also a slightly simpler way in letting the function themselves index into the prototype, like

```ts
function addDays() { ... }
Date.prototype[addDays] = addDays
```

...but TS currently complains about that since `addDays` is not a `symbol` or `string` or `number`

**Comment by staeke** [comment]
and to answer your question @arturohernandez10 - sure go ahead with a PR!
